### PR TITLE
checker: improve failover speed with witness

### PR DESCRIPTION
**Pull request by ethercflow**
ref tikv/pd#5627

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5627

### What is changed and how does it work?

```commit-message
checker: improve failover speed with witness
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

Code changes

- None

Side effects

- None

Related changes

- None

### Release note

```release-note
None
```

### CODE REVIEW

File: server/schedule/operator/operator.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ type Operator struct {
 func NewOperator(desc, brief string, regionID uint64, regionEpoch *metapb.RegionEpoch, kind OpKind, approximateSize int64, steps ...OpStep) *Operator {
 	level := core.Medium
 	if kind&OpAdmin != 0 {
-		level = core.Urgent
+		level = core.High
```
Comment (Connor1996) [inline server/schedule/operator/operator.go:60]: it may not be necessary

File: server/schedule/checker/rule_checker_test.go [hunk 2]
```diff
@@ -124,8 +124,8 @@ func (suite *ruleCheckerTestSuite) TestFixPeer() {
 	r = r.Clone(core.WithDownPeers([]*pdpb.PeerStats{{Peer: r.GetStorePeer(2), DownSeconds: 60000}}))
 	op = suite.rc.Check(r)
 	suite.NotNil(op)
-	suite.Equal("replace-rule-down-peer", op.Desc())
-	suite.Equal(core.High, op.GetPriorityLevel())
+	suite.Equal("replace-rule-fast-failover-peer", op.Desc())
```
Comment (rleungx) [inline server/schedule/checker/rule_checker_test.go:127]: How about keeping the `down-peer`?

**Comment by ethercflow** [comment]
image

After testing several times, I found that it takes 1min, and I will continue to see how to compress it to 30s.

**Comment by ethercflow** [comment]
image

File: server/schedule/checker/rule_checker.go [hunk 3]
```diff
@@ -233,16 +233,17 @@ func (c *RuleChecker) fixRulePeer(region *core.RegionInfo, fit *placement.Region
 func (c *RuleChecker) addRulePeer(region *core.RegionInfo, rf *placement.RuleFit) (*operator.Operator, error) {
 	ruleCheckerAddRulePeerCounter.Inc()
 	ruleStores := c.getRuleFitStores(rf)
-	store, filterByTempState := c.strategy(region, rf.Rule).SelectStoreToAdd(ruleStores)
+	isWitness := rf.Rule.IsWitness
```
Comment (bufferflies) [inline server/schedule/checker/rule_checker.go:236]:
```suggestion
	isWitness := rf.Rule.IsWitness&&c.isWitnessEnabled()
```

File: server/schedule/filter/filters.go [hunk 4]
```diff
@@ -495,6 +498,8 @@ func (f *StoreStateFilter) anyConditionMatch(typ int, opt *config.PersistOptions
 		funcs = []conditionFunc{f.isRemoved, f.isRemoving, f.isDown, f.isDisconnected, f.isBusy}
 	case scatterRegionTarget:
 		funcs = []conditionFunc{f.isRemoved, f.isRemoving, f.isDown, f.isDisconnected, f.isBusy}
+	case fastFailoverTarget:
+		funcs = []conditionFunc{f.isRemoved, f.isRemoving, f.isDown, f.isDisconnected}
```
Comment (rleungx) [inline server/schedule/filter/filters.go:502]: Should we include `isBusy`?

**Comment by ethercflow** [comment]
28d2e7ae-b7ed-4dd8-a1c0-3d1f83ff0143
257868ef-b241-46ab-8eab-188c56146a93

The current test failed to reproduce the long-tail problem encountered before, and the current test results are in line with expectations.

**Review by rleungx — APPROVED** [review]

**Review by bufferflies — APPROVED** [review]

**Pull request merged**